### PR TITLE
Move dev-console and knative dependencies out of console-shared

### PR DESCRIPTION
**Pull request by vikram-raj**
**Story**: 
https://issues.redhat.com/browse/ODC-2068

**Description**: 
The `console-shared` package should not depend on any other package. Remove the `dev-console` and `knative-plugin` package dependencies from `console-shared`.

### CODE REVIEW

File: frontend/packages/console-shared/src/utils/utils.ts [hunk 1]
```diff
@@ -76,3 +80,31 @@ export const toTitleCase = (title: string): string => {
 export const isModifiedEvent = (event: React.MouseEvent<HTMLElement>) => {
   return !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
 };
+
+const catalogItemCompare = (keyword: string, item: CatalogItem): boolean => {
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/utils/utils.ts:84]: I think these utils are specific to catalog and should be co-located with it. If needed we can pull the generic catalog pieces out into `@console/shared` or maybe even SDK. @christianvogt WDYT?

File: frontend/packages/console-shared/src/utils/__tests__/resource-utils.spec.ts [hunk 2]
```diff
@@ -69,11 +51,7 @@ const knativeKeys = [Keys.REVISIONS, Keys.KNATIVECONFIGS, Keys.KSROUTES];
 
 describe('TransformResourceData', () => {
   it('should create Deployment config Items', () => {
-    const transformedData = createOverviewItemsForType(
-      'deploymentConfigs',
-      MockResources,
-      knativeOverviewResourceUtils,
-    );
+    const transformedData = createOverviewItemsForType('deploymentConfigs', MockResources);
```
Comment (rohitkrai03) [inline frontend/packages/console-shared/src/utils/__tests__/resource-utils.spec.ts:54]: So now this is not testing the knative utils scenario anymore?

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/packages/console-shared/src/utils/__tests__/resource-utils.spec.ts (hunk 2 above)
Comment (vikram-raj) [reply frontend/packages/console-shared/src/utils/__tests__/resource-utils.spec.ts]: Removed the `createOverviewItemsForType` util as it is not used anywhere.

**Comment by vikram-raj** [comment]
Propagate from epic:
/label docs-approved
/label px-approved

**Review by jerolimov — APPROVED** [review]
Noticed that there are still three links from console-shared to knative-plugin models.

image

Discussed this with @vikram-raj, we expect that this will be migrated when switching to dynamic plugin SDK.

/lgtm

**Comment by jerolimov** [comment]
Tested this on a cluster with cluster bot and works there fine as well.

/label qe-approved

**Pull request merged**